ospfd/ospf_network.c

```diff
@@ -190,7 +190,7 @@ int ospf_sock_init(struct ospf *ospf)
 			flog_err(EC_LIB_SOCKET,
 				 "ospf_read_sock_init: socket: %s",
 				 safe_strerror(errno));
-			exit(1);
+			return -1;
```

Comment:
This commit is really 2 different commits imo.  1 -> don't exit and just return and then the closing of the socket when the kernel interactions go south.  Otherwise looks good

Reply:
I removed the "close" part. I initially thought it was a leak but actually, it may be a deliberate decision to continue working even when setsockopt error occurs. In other daemons, we also don't stop on a setsockopt error.